sphinx/source/docs/user_guide/styling.rst

```diff
@@ -326,6 +305,12 @@ Take a 400 pixel by 400 pixel graph with x and y axes ranging from 0
 through 10, for example. A glyph that is one fifth as wide and tall as the graph
 would have a size of 80 screen units or 2 data-space units.
 
+Many objects in Bokeh have a dedicated property that you can set to either
```

Comment:
I noticed that some objects (such as `location_units` of the Spans annotation) have an explicit property to switch from `data` to `screen` units. While other objects (such as the Whiskers annotation) do not have an explicit property for this but still accept both `data` and `screen` units. Is there a reason for this difference? Anything I could add to this paragraph for additional clarity?

Reply:
Anything that is a `PropertyUnitsSpec` also implicitly, automatically defines a corresponding separate `_units` property to set the units (so the version below with `some_data` and a dict is never necessary). You can see that `Whisker` does in fact accept the `_units` properties here:
https://github.com/bokeh/bokeh/blob/18f3e470c923a9838c7528903beb6ecae03fb3db/examples/integration/annotations/whisker.py#L24-L27

I think it is simply a flaw in the automated ref guide generation that these automatic location properties are not picked up in the generated docs. If something has an *explicit* `_units` then that is typically because the corresponding property (e.g. `location`) is not a dataspec. The automatic `_units` properties are only generated for `PropertyUnitsSpec`.